src/overlays/actors/ovl_En_Snowwd/z_en_snowwd.c

```diff
@@ -30,25 +30,109 @@ ActorInit En_Snowwd_InitVars = {
+void EnSnowwd_Init(Actor* thisx, PlayState* play) {
+    EnSnowwd* this = THIS;
+
+    this->actor.home.rot.z = 0;
+    this->actor.home.rot.y = 0;
+    this->unk190 = 0;
+    this->actor.uncullZoneForward = 4000.0f;
+    this->actor.uncullZoneScale = 2000.0f;
+    this->actor.uncullZoneDownward = 2400.0f;
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actionFunc = func_80AF76F0;
+}
+
+void EnSnowwd_Destroy(Actor* thisx, PlayState* play) {
+    EnSnowwd* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void func_80AF76F0(EnSnowwd* this, PlayState* play) {
+    s32 pad;
+    Actor* thisx = &this->actor;
+    f32 sp54;
+    Vec3f sp48;
+
+    if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        Actor_PlaySfx(thisx, NA_SE_IT_REFLECTION_WOOD);
+    }
+    if (thisx->home.rot.y != 0) {
+        this->unk190 = 0x15;
+        thisx->home.rot.y = 0;
+        if (thisx->home.rot.z == 0) {
+            if (SNOWWD_GET_PARAM_F80(&this->actor) < 0x10) {
+                sp48 = thisx->world.pos;
+                sp48.y += 200.0f;
+                Item_DropCollectibleRandom(play, NULL, &sp48, (SNOWWD_GET_PARAM_F80(&this->actor) * 0x10));
+            }
+            thisx->home.rot.z = 1;
+        }
+    }
+    if (thisx->xzDistToPlayer < 600.0f) {
+        Collider_UpdateCylinder(thisx, &this->collider);
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    }
+    if (this->unk190 > 0) {
+        this->unk190--;
+        sp54 = Math_SinS((s16)((this->unk190 ^ 0xFFFF) * 0x3332)) * 250.0f;
+        thisx->shape.rot.x = (s16)(Math_CosS(thisx->yawTowardsPlayer - thisx->shape.rot.y) * sp54);
+        thisx->shape.rot.z = (s16)(Math_SinS(thisx->yawTowardsPlayer - thisx->shape.rot.y) * sp54);
+        sp48 = thisx->world.pos;
+        sp48.x += Rand_CenteredFloat(80.0f);
+        sp48.y += 100.0f + Rand_ZeroFloat(30.0f);
+        sp48.z += Rand_CenteredFloat(80.0f);
+        func_800B0EB0(play, &sp48, &D_80AF7AC8, &D_80AF7ABC, &D_80AF7AD4, &D_80AF7AD8, 200, 10, 20);
+    }
+}
+
+void EnSnowwd_Update(Actor* thisx, PlayState* play) {
+    EnSnowwd* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Snowwd/EnSnowwd_Destroy.s")
+    this->actionFunc(this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Snowwd/func_80AF76F0.s")
+void EnSnowwd_Draw(Actor* thisx, PlayState* play) {
+    OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Snowwd/EnSnowwd_Update.s")
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    Gfx_SetupDL25_Opa(play->state.gfxCtx);
+    gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(&D_06001AA0));
```

Comment:
```suggestion
    gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(object_snowwd_Tex_001AA0));
```

So looking at `object_snowwd.xml` I see that this particular offset is left as a commented out blob, which basically means we were unable to determine what it actually was when the object files were auto generated. However as we see here it gets placed in segment 8 (if you are curious about segments see https://github.com/zeldaret/mm/blob/master/docs/tutorial/data.md#segmented-pointers), with a display list used afterwards. With that information we can determine what this actually as and update the xml so we can use the correct symbol here.

To do that we look for `object_snowwd_DL_000198` in `object_snowwd.c`, and look for `0x08000000` (this is segment 8). From there we see that it is used in the `gsDPLoadTextureBlock` command, meaning `D_06001AA0` is a texture and we can read all the information needed to update the xml with. In particular we know it is an RGBA16 that is 32x64 pixels wide. 

Thus we can replace the commented out blob with:

`<Texture Name="object_snowwd_Tex_001AA0" OutName="tex_001AA0" Format="rgba16" Width="32" Height="64" Offset="0x1AA0" />` 

see https://github.com/zeldaret/mm/blob/master/tools/ZAPD/docs/zapd_extraction_xml_reference.md if you want to learn more about xmls and there formats.
After re-extracting, you would know be able to replace `&D_06001AA0` with `object_snowwd_Tex_001AA0`.

If you have any questions feel free to ask in the discord (you can ping me directly if you want, I am Anon58 :) ).

Reply:
The other blob doesn't get used, but I confirmed with Texture64 that it has the exact same attributes as the previous one. So if you wouldn't mind replacing that blob as well with

`<Texture Name="object_snowwd_Tex_002A0" OutName="tex_002A0" Format="rgba16" Width="32" Height="64" Offset="0x2A0" />`

That would be much appreciated :) .